### ISSUE TITLE
Support for music.youtube.com

### ISSUE THREAD

**Issue by redsx16** [web-scrobbler/web-scrobbler#1641]
Music.youtube.com is not available as a connector. I tried adding https://music.youtube.com/* and *.youtube.com/ as url patterns. The it now shows it as supported but does not detect any music playing so nothing is scrobbling. FYI: This is the new Youtube Music that will eventually replace Google Play Music.

### Expected behavior
Songs scrobble to last.fm with proper artist and song title.

### Current behavior
None of the tracks playing are recognized so nothing scrobbles.

### Possible solution
Since the site is new most likely a new connector is needed to match formatting.

### Steps to reproduce (for bugs)

https://pastebin.mozilla.org/9086861

1. Load up music.youtube.com
2. Play any song
3. Check Last.fm or the extension for status
4. Try pressing bottom right corner arrow and notification pops indicating track not recognized

### Steps for resolving the issue

- [ ] Reauthenticate on Last.fm

- [ ] Try to use unpacked extension from latest master

- [ ] Check other issues for duplicates

### Environment

* Version used: 2.5 from Chrome store
* Browser name and version: Chrome 67.0.3396.62 (Official Build) beta (64-bit)

**Comment by jawshoeah** [comment]
I can confirm music.youtube.com is not scrobbling.

**Comment by jaccarmac** [comment]
Just got my invite so I will attempt to whip up a connector tonight.

**Comment by jaccarmac** [comment]
Implemented, please give the PR a try if you feel comfortable running from source. I could use tests from a non-American locale, and the code is very rough but functional on my end!

**Comment by redsx16** [comment]
So far it's working for me. Only issue I've noticed is that songs that have a "feat" artist are also found in the artist field when it's scrobbling with Force Track Recognition enabled. Each case the "feat" artist is after the main artist, ex: BlocBoy JB & Drake — Look Alive (feat. Drake) should be BlocBoy JB — Look Alive (feat. Drake)

**Comment by jaccarmac** [comment]
Thanks for giving it a go. I noticed that. Gonna wait on making a change, as personally I tend to like the Beets style of scrobbling (features go in the artist field instead). Additionally check out a track that's supposed to have multiple artists on it, like https://music.youtube.com/watch?v=6qwwuKIj8BA.

**Comment by redsx16** [comment]
Your example makes sense because there is no featured artist so none of the artists are found in both the artist field and the title field. At least for me it's unnecessary to have it duplicated. I don't know what the standards are.

**Comment by jaccarmac** [comment]
I'm with you, the deduplication is less trivial than getting the website to scrobble, so it may be better for another ticket. @inverse what's your view on that?

**Comment by GintokiHub** [comment]
I am having the same issue, as  I am not sure where to put the folder with the extension to install it manually in vivaldi (chromium based), I haven't been able to test it out with possibly(?) the newer version of your app.

Here is a link to the youtube tune I was listening to: https://www.youtube.com/watch?v=o1gN2qE8hNM

**Comment by djwesterfield** [comment]
I'm having the same issue with the updated scrobbler extention and Chrome.

**Comment by jaccarmac** [comment]
Are you running unpacked from my branch? Code's not in the upstream repository yet. cc @GintokiHub + @djwesterfield

**Comment by rlcx** [comment]
Hello, it look like even the unpacked version from your branch is not working. At least, not here.
My locale is fr-FR.

**Comment by jaccarmac** [comment]
See anything interesting in the console on the background page?

**Comment by inverse** [comment]
@rlcx could be due to the way the play button is deleted... I guess it works in `en-*` locales only

**Comment by jaccarmac** [comment]
Ah yes, since I'm checking the text for state. In non EN locales are there any other properties on the button I could check? Looks like I may need to reconsider that bit.

**Comment by thwaller** [comment]
Is there an estimated time frame for this to be added? I tried doing the manual add and it still said unsupported, so rather than messing something up, I am waiting for this to get into the actual release.

**Comment by jaccarmac** [comment]
Strange that you can't get it to work.

The patch as written works for me so I've put it on my back burner for now. Want to polish it and get it merged but I'd like to do it right (fixing the way artists are parsed for both this and TIDAL). So no hard ETA from my end.

**Comment by thwaller** [comment]
If it works, why would you not make it "live" so others can use it, at least as it is?

**Comment by jaccarmac** [comment]
The pull request is open, the owner wants me to improve it before they accept it.

**Comment by AntwerpPeter** [comment]
Any news on when it will be available in the release version?
I am running chrome with a be/nl locale.

**Comment by lelukas** [comment]
guys, so, any news about the fix for this issue?

**Comment by inverse** [comment]
@lelukas the open PR needs a bit of work from @jaccarmac before it'll be merged.

**Comment by sukalov** [comment]
waiting the moment, you start supporting youtube music, to start using it!

**Comment by inverse** [comment]
The PR has been merged. Unfortunately I don't have release power :/

**Comment by jaccarmac** [comment]
Let's close this since it got deployed. We can track bugs in separate tickets like the one above.

**Comment by inverse** [comment]
:+1:
